Apps/Admin/Tests/Functional/cypress/support/functions/intercept.js

```diff
@@ -0,0 +1,128 @@
+        case "/feedback":
+            setupFeedbackAliases();
+            break;
+        case "/beta-access":
+            setupBetaAccessAliases();
+            break;
+        case "/reports":
+            setupReportAliases();
+            break;
+        default:
+            cy.log(`Alias setup on login not required for page: ${page}`);
+            break;
+    }
+}
+
+export function waitForInitialDataLoad(page) {
+    switch (page) {
+        case "dashboard":
+            waitForDashboard();
+            break;
+        case "communications":
+            waitForCommunication();
+            break;
+        case "feedback":
+            waitForFeedback();
+            break;
+        case "beta-access":
+            waitForBetaAccess();
+            break;
+        case "reports":
+            waitForReport();
+            break;
+        default:
+            cy.log(
+                `Wait for initial data load on login not required for page: ${page}`
+            );
+            break;
+    }
+}
+
+function setupDashboardAliases() {
+    cy.log("Setting up dashboard aliases.");
+
+    cy.intercept("GET", "**/Dashboard/AllTimeCounts").as("getAllTimeCounts");
+    cy.intercept("GET", `**/Dashboard/AppLoginCounts*`).as("getAppLoginCounts");
+    cy.intercept("GET", "**/Dashboard/Ratings/Summary*").as(
+        "getRatingsSummary"
+    );
+    cy.intercept("GET", "**/Dashboard/AgeCounts*").as("getAgeCounts");
+    cy.intercept("GET", "**/Dashboard/DailyUsageCounts*").as(
+        "getDailyUsageCounts"
+    );
+    cy.intercept("GET", "**/Dashboard/RecurringUserCount*").as(
+        "getRecurringUserCount"
+    );
+}
+
+function setupCommunicationAliases() {
+    cy.log("Setting up communication aliases.");
+
+    cy.intercept("GET", "**/Broadcast/").as("getBroadcast");
+    cy.intercept("GET", `**/Communication/*`).as("getCommunication");
+}
+
+function setupFeedbackAliases() {
+    cy.log("Setting up feedback aliases.");
+
+    cy.intercept("GET", "**/Tag/").as("getTag");
+    cy.intercept("GET", `**/UserFeedback/`).as("getUserFeedback");
+}
+
+function setupBetaAccessAliases() {
+    cy.log("Setting up beta access aliases.");
+
+    cy.intercept("GET", "**/BetaFeature/AllUserAccess").as("getAllUserAccess");
+}
+
+function setupReportAliases() {
+    cy.log("Setting up report aliases.");
+
+    cy.intercept("GET", "**/AdminReport/BlockedAccess").as("getBlockedAccess");
+    cy.intercept("GET", "**/AdminReport/ProtectedDependents*").as(
+        "getProtectedDependents"
+    );
+}
+
+function waitForDashboard() {
+    cy.log("Wait on dashboard");
+    cy.wait("@getAllTimeCounts", { timeout: defaultTimeout });
+    cy.wait("@getAppLoginCounts", { timeout: defaultTimeout });
+    cy.wait("@getRatingsSummary", { timeout: defaultTimeout });
+    cy.wait("@getAgeCounts", { timeout: defaultTimeout });
+    cy.wait("@getDailyUsageCounts", { timeout: defaultTimeout });
+    cy.wait("@getRecurringUserCount", { timeout: defaultTimeout });
```

Comment:
I wanted to figure out exactly how the `wait()` commands work here.

Our front-end makes these requests in parallel, so it's extremely unlikely we would receive a response from the back-end for the first request before the front-end issues the second request (and so on).  My understanding is that, like other commands, each `wait()` bottlenecks the test so it won't proceed to the next command until the current one finishes.

The documentation states there are 2 different timeout values used by `wait()`:

- `requestTimeout`: Time, in milliseconds, to wait for a request to go out
- `responseTimeout`: Time, in milliseconds, to wait until a response

By passing a 60000ms value to the `timeout` property, we are overriding both of these.

I would expect the test to hit line 99, start a timer for 60000ms (the overridden `requestTimeout`), pause the test until the site issues a matching HTTP request for `@getAllTimeCounts`, then start another timer for 60000ms (the overridden `responseTimeout`), pausing until the server sends an HTTP response.  If the request or response do not happen in the given time frames, the test should fail.

However, we are not issuing our requests in between each of these `wait()` commands.  Therefore, if the tests are passing, the window for making the request must not start when the `wait()` is issued - it must start when the `intercept()` is issued.  I couldn't find this described in the documentation when I looked.  It may be worth playing around with timeouts to be absolutely certain of this.

![image](https://github.com/bcgov/healthgateway/assets/16570293/a021dd71-2c2b-4372-8035-db1ce759f328)

Putting all that aside, if we are issuing requests simultaneously, I think it would make more sense to use a single `wait()`, passing in each of the aliases in an array.  That way, it ensures we wait a maximum of 60s for all of the responses instead of waiting 60s for the first, waiting another 60s for the second, etc.

Reply:
Changed to array.

<img width="524" alt="Screenshot 2024-07-05 at 9 30 03 AM" src="https://github.com/bcgov/healthgateway/assets/58790456/20dbdd1b-19b7-4988-bff4-8af76f52518d">
<img width="492" alt="Screenshot 2024-07-05 at 9 30 19 AM" src="https://github.com/bcgov/healthgateway/assets/58790456/007719f8-a4f0-4a66-a6c5-5a1d94001e09">
<img width="504" alt="Screenshot 2024-07-05 at 9 30 47 AM" src="https://github.com/bcgov/healthgateway/assets/58790456/64ddfcc9-18b8-4771-b1d4-93ea8204ec3b">